### PR TITLE
Adds permutation utilities to ParityPreservingFermionicGate [Part of #353]

### PR DESCRIPTION
**Pull request by bryano**

### CODE REVIEW

**Comment by bryano** [comment]
@kevinsung ping

File: openfermioncirq/gates/fermionic_simulation.py [hunk 1]
```diff
@@ -140,6 +140,11 @@ def fermion_generator_components(self
         generator is :math:`\sum_i w_i G_i + \text{h.c.}` where :math:`(w_i)_i`
         are the gate's weights."""
 
+    @abc.abstractmethod
+    def fswap(self, i: int):
+        """Updates the weights of the gate as if it were sandwiched by an FSWAP
+        on the i-th and (i+1)th qubits."""
```
[the remark this answers is not in the document]
Comment (bryano) [reply openfermioncirq/gates/fermionic_simulation.py]: Done

[the remark this answers is not in the document]
Comment (bryano) [reply openfermioncirq/gates/fermionic_simulation.py]: Should I merge?

[the remark this answers is not in the document]
Comment (kevinsung) [reply openfermioncirq/gates/fermionic_simulation.py]: Yeah feel free to merge after approval.

**Pull request merged**